lib/ordered_set.dart

```diff
@@ -48,7 +48,7 @@ class OrderedSet<E> extends IterableMixin<E> implements Iterable<E> {
 
   @override
   Iterator<E> get iterator {
-    return _OrderedSetIterator<E>(this);
+    return _backingSet.expand<E>((es) => es).iterator;
```

Comment:
nice, didn't know about expand!

Reply:
will this guarantee the order?

Reply:
as far as I see in the code it does almost the same `_OrderedSetIterator` except it is already supporting null safety and it behaves close to the iterator protocol in some cases, i.e. before the first `moveNext` call